cdap-docs/developers-manual/source/data-exploration/object-mapped-tables.rst

```diff
@@ -0,0 +1,118 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _object-mapped-table-exploration:
+
+============================================
+ObjectMappedTable Exploration
+============================================
+
+An ObjectMappedTable is a system Dataset that can write java objects to a Table
+by mapping object fields to Table columns. It can also be explored in an ad-hoc manner.
+
+Creating an ObjectMappedTable
+-----------------------------
+
+When creating an ``ObjectMappedTable`` in your application, you must specify the Java type
+that will be stored in your table::
+
+  @Override
+  public void configure() {
+    try {
+      createDataset("purchases", ObjectMappedTable.class,
+                    ObjectMappedTableProperties.builder()
+                      .setType(Purchase.class)
+                      .build()
+                   );
+    } catch (UnsupportedTypeException e) {
+      // This exception is thrown by ObjectMappedTable if its parameter type cannot be
+      // (de)serialized (for example, if it is an interface and not a class, then there is
+      // no auto-magic way deserialize an object.) In this case that will not happen
+      // because Purchase is an actual classes.
+      throw new RuntimeException(e);
+    }
+  } 
+
+CDAP will derive the record schema from that type. For example, if the ``Purchase`` class is defined as::
+
+  public class Purchase {
+    private final String customer, product;
+    private final int quantity, price;
+    private final long purchaseTime;
+    private String catalogId;
+
+    public Purchase(String customer, String product, int quantity, int price, long purchaseTime) {
+      this.customer = customer;
+      this.product = product;
+      this.quantity = quantity;
+      this.price = price;
+      this.purchaseTime = purchaseTime;
+      this.catalogId = "";
+    }
+  }
+
+CDAP will map each object field to a table column and the schema will be::
+
+  (key BINARY, catalogid STRING, customer STRING, price INT, product STRING, purchasetime BIGINT, quantity INT))
+
+Note that column names are all changed to be lowercase. This is because Hive column names are case insensitive.
```

Comment:
Note that all column names have been changed to lowercase letters. This is because Hive column names are case-insensitive. In addition to the object fields, the object key has been inserted into the schema as a binary column ``key``.

If you wish to name the key column differently, perhaps because your object already contains a field named "key", you 
can do so by providing an additional property when creating your Dataset. You can also set the key type, such as to
a ``String``, if desired::